### PR TITLE
Fix/1125 encapsulate all preview generator error for content endpoint

### PR DESCRIPTION
**Pull request by inkhey**
related to tracim/tracim#1125

### CODE REVIEW

File: backend/tracim_backend/lib/core/content.py [hunk 1]
```diff
@@ -1583,7 +1586,7 @@ def get_preview_page_nb(self, revision_id: int, file_extension: str) -> typing.O
                 file_path,
                 file_ext=file_extension
             )
-        except UnsupportedMimeType:
+        except Exception:
```
Comment (buxx) [inline backend/tracim_backend/lib/core/content.py]: I suggest to continue to except `UnsupportedMimeType` (and return None in that case) and except Exception (and log a warning about the error).

File: backend/tracim_backend/lib/core/content.py [hunk 2]
```diff
@@ -1594,7 +1597,7 @@ def has_pdf_preview(self, revision_id: int, file_extension: str) -> bool:
...
```
Comment (buxx) [inline backend/tracim_backend/lib/core/content.py]: I suggest to continue to except `UnsupportedMimeType` (and return None in that case) and except Exception (and log a warning about the error).

File: backend/tracim_backend/lib/core/content.py [hunk 3]
```diff
@@ -1605,7 +1608,7 @@ def has_jpeg_preview(self, revision_id: int, file_extension: str) -> bool:
...
```
Comment (buxx) [inline backend/tracim_backend/lib/core/content.py]: I suggest to continue to except `UnsupportedMimeType` (and return None in that case) and except Exception (and log a warning about the error).

File: backend/tracim_backend/lib/core/content.py [hunk 4]
```diff
@@ -955,9 +954,11 @@ def get_pdf_preview_path(
                 page=page_number,
                 file_ext=file_extension,
             )
+        except PageOfPreviewNotFound as exc:
+            raise exc
         except UnavailablePreviewType as exc:
             raise TracimUnavailablePreviewType() from exc
-        except UnsupportedMimeType as exc:
+        except Exception as exc:
```
Comment (buxx) [inline backend/tracim_backend/lib/core/content.py]: I suggest to continue to except `UnsupportedMimeType` (and raise UnavailablePreview with same message ) and except Exception (and raise UnavailablePreview with message about unkown error).

File: backend/tracim_backend/lib/core/content.py [hunk 5]
```diff
@@ -975,7 +976,7 @@ def get_full_pdf_preview_path(self, revision_id: int, file_extension: str) -> st
             pdf_preview_path = self.preview_manager.get_pdf_preview(file_path, file_ext=file_extension)  # nopep8
         except UnavailablePreviewType as exc:
             raise TracimUnavailablePreviewType() from exc
-        except UnsupportedMimeType as exc:
+        except Exception as exc:
```
Comment (buxx) [inline backend/tracim_backend/lib/core/content.py]: I suggest to continue to except `UnsupportedMimeType` (and raise UnavailablePreview with same message ) and except Exception (and raise UnavailablePreview with message about unkown error).

File: backend/tracim_backend/lib/core/content.py [hunk 6]
```diff
@@ -1045,7 +1046,9 @@ def get_jpg_preview_path(
                 height=height,
                 file_ext=file_extension,
             )
-        except UnsupportedMimeType as exc:
+        except (PreviewDimNotAllowed, PageOfPreviewNotFound) as exc:
+            raise exc
```
Comment (buxx) [inline backend/tracim_backend/lib/core/content.py]: If this except is only for reader understanding, can you add a comment to explain why these exceptions can be throwed ?

File: backend/tracim_backend/lib/core/content.py [hunk 7]
```diff
@@ -1045,7 +1046,9 @@ def get_jpg_preview_path(
...
+        except Exception as exc:
```
Comment (buxx) [inline backend/tracim_backend/lib/core/content.py]: In that case it is preferable to explain in message what we caught an unkown error

File: backend/tracim_backend/lib/core/content.py (hunk 5 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: done

File: backend/tracim_backend/lib/core/content.py (hunk 4 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: done

File: backend/tracim_backend/lib/core/content.py (hunk 6 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: done

File: backend/tracim_backend/lib/core/content.py (hunk 7 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: done

File: backend/tracim_backend/lib/core/content.py [hunk 8]
```diff
@@ -1586,31 +1619,50 @@ def get_preview_page_nb(self, revision_id: int, file_extension: str) -> typing.O
...
-            has_preview = False
-        return has_preview
+        except UnsupportedMimeType:
+            return False
+        except Exception as e:
+            logger.warning(
+                self,
+                "Unknown Preview_Generator Exception Occured : {}".format(str(e))
+            )
+            logger.debug(self, traceback.format_exc())
```
Comment (lebouquetin) [inline backend/tracim_backend/lib/core/content.py]: je mettrais la traceback dans les logs warning pour l'avoir aussi en prod

File: backend/tracim_backend/lib/core/content.py [hunk 9]
```diff
@@ -1586,31 +1619,50 @@ def get_preview_page_nb(self, revision_id: int, file_extension: str) -> typing.O
...
```
Comment (lebouquetin) [inline backend/tracim_backend/lib/core/content.py]: même remarque

File: backend/tracim_backend/lib/core/content.py [hunk 10]
```diff
@@ -1586,31 +1619,50 @@ def get_preview_page_nb(self, revision_id: int, file_extension: str) -> typing.O
...
             )
-        except Exception:
+        except UnsupportedMimeType:
+            return None
+        except Exception as e:
+            logger.warning(
+                self,
+                "Unknown Preview_Generator Exception Occured : {}".format(str(e))
+            )
+            logger.debug(self, traceback.format_exc())
```
Comment (lebouquetin) [inline backend/tracim_backend/lib/core/content.py]: même remarque (logger.warning)

File: backend/tracim_backend/lib/core/content.py [hunk 11]
```diff
@@ -1047,8 +1069,19 @@ def get_jpg_preview_path(
...
+        except UnsupportedMimeType as exc:
+            raise UnavailablePreview(
+                'No preview available for content {}, revision {}'.format(content_id, revision_id)  # nopep8
+            ) from exc
         except Exception as exc:
+            logger.warning(
+                self,
+                "Unknown Preview_Generator Exception Occured : {}".format(str(exc))
+            )
+            logger.debug(self, traceback.format_exc())
```
Comment (lebouquetin) [inline backend/tracim_backend/lib/core/content.py]: logger.warning

File: backend/tracim_backend/lib/core/content.py [hunk 12]
```diff
@@ -976,7 +989,16 @@ def get_full_pdf_preview_path(self, revision_id: int, file_extension: str) -> st
...
+        except UnsupportedMimeType as exc:
+            raise UnavailablePreview(
+                'No preview available for revision {}'.format(revision_id)
+            ) from exc
         except Exception as exc:
+            logger.warning(
+                self,
+                "Unknown Preview_Generator Exception Occured : {}".format(str(exc))
+            )
+            logger.debug(self, traceback.format_exc())
```
Comment (lebouquetin) [inline backend/tracim_backend/lib/core/content.py]: logger.warning

File: backend/tracim_backend/lib/core/content.py [hunk 13]
```diff
@@ -955,10 +957,21 @@ def get_pdf_preview_path(
...
+        except UnsupportedMimeType as exc:
+            raise UnavailablePreview(
+                'No preview available for content {}, revision {}'.format(content_id, revision_id)  # nopep8
+            ) from exc
         except Exception as exc:
+            logger.warning(
+                self,
+                "Unknown Preview_Generator Exception Occured : {}".format(str(exc))
+            )
+            logger.debug(self, traceback.format_exc())
```
Comment (lebouquetin) [inline backend/tracim_backend/lib/core/content.py]: logger.warning

File: backend/tracim_backend/lib/core/content.py (hunk 8 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: fait

File: backend/tracim_backend/lib/core/content.py (hunk 9 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: fait

File: backend/tracim_backend/lib/core/content.py (hunk 10 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: fait

File: backend/tracim_backend/lib/core/content.py (hunk 11 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: fait

File: backend/tracim_backend/lib/core/content.py (hunk 12 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: fait

File: backend/tracim_backend/lib/core/content.py (hunk 13 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: fait

**Comment by buxx** [comment]
@inkhey News about 3 remaining comments ?

File: backend/tracim_backend/lib/core/content.py (hunk 1 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: done

File: backend/tracim_backend/lib/core/content.py (hunk 2 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: done

File: backend/tracim_backend/lib/core/content.py (hunk 3 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: done

**Pull request merged**